python/tk_multi_publish2/dialog.py

```diff
@@ -319,35 +321,32 @@ def _update_task_details_ui(self, new_task_selection=None):
             # Note: At this point we don't really care if current task actually had a UI, we can
             # certainly tear down an empty widget.
             logger.debug("The ui is going to change, so clear the current one.")
-            self.ui.custom_settings_page.widget = None
+            self.ui.task_settings.widget = None
             self._current_tasks = new_task_selection
             return
 
         # A task was picked, so make sure our page is in foreground.
         self.ui.details_stack.setCurrentIndex(self.TASK_DETAILS)
 
+        # set the header for the task plugin
         self.ui.task_icon.setPixmap(new_task_selection.plugin.icon)
         self.ui.task_name.setText(new_task_selection.plugin.name)
 
-        # If the new task selection does not have a custom UI, a simple tear down of the custom UI
-        # and setting the built-in fields will suffice.
         if not new_task_selection.has_custom_ui:
             logger.debug("Clearing custom UI and using default task details...")
-            self.ui.custom_settings_page.widget = None
 
-            # All the items are of the same type,
-            self.ui.settings_stack.setCurrentIndex(self.BUILTIN_TASK_DETAILS)
-            self.ui.task_description.setText(new_task_selection.plugin.description)
+            # a bit of a hack here, but the base hook requires a description in
+            # order to function. we'll simply monky patch it here to be able
+            # to generate the UI.
+            self._plugin_base_hook.description = new_task_selection.plugin.description
```

Comment:
not sure i totally follow, but could this be done via inheritance? doesn't `new_task_selection.plugin.description` inherit from plugin base hook?

Reply:
This is hopefully true, but for studios that haven't updated core, that base class injection will fail silently, and the regular `Hook` base class will be used. That means the hook may not have the custom UI. So we keep this reusable hook instance that does implement the methods and we use that to generate the widget. It requires a description, so we essentially monkey patch it with the selected tasks' description.

Reply:
but if they are using `HookBaseClass = sgtk.get_hook_baseclass()` they will just get this new interface auto-injected, right? Sorry, i am probably missing some detail here! :)

Reply:
Only if they have the new core. At least the way we're currently doing the injection. Lets discuss.